src/main/java/org/reactome/addlinks/dataretrieval/ensembl/EnsemblBiomartRetriever.java

```diff
@@ -0,0 +1,232 @@
+        // Species names are in biomart format (eg: hsapiens).
+        for (String speciesName : EnsemblBiomartUtil.getSpeciesNames()) {
+            String speciesBiomartName = EnsemblBiomartUtil.getBiomartSpeciesName(speciesName);
+            logger.info("Retrieving Biomart files for " + speciesBiomartName);
+
+            logger.info("Retrieving microarray data");
+            // Query Biomart for existing microarray 'types' (not ids) that exist for this species.
+            Set<String> microarrayTypes = queryBiomart(getMicroarrayTypesQuery(speciesBiomartName), 0);
+            // Iterate through each microarray type and retrieve Ensembl-Microarray identifier mappings.
+            // All mappings are stored in a single file, (eg: hsapiens_microarray);
+            for (String microarrayType : microarrayTypes) {
+                queryBiomartAndStoreData(speciesBiomartName, getBiomartXMLFilePath(), microarrayType, "_microarray");
+            }
+
+            // Query Ensembl-Uniprot (swissprot and trembl) identifier mapping data from Biomart
+            // and write it to a file (eg: hsapiens_uniprot).
+            logger.info("Retrieving UniProt data");
+            for (String uniprotQueryId : Arrays.asList("uniprotswissprot", "uniprotsptrembl")) {
+                queryBiomartAndStoreData(speciesBiomartName, getBiomartXMLFilePath(), uniprotQueryId, "_uniprot");
+            }
+
+            logger.info("Completed Biomart data retrieval for " + speciesBiomartName);
+        }
+    }
+
+    // Updates Biomart XML Query with Microarray type information.
+    private String getMicroarrayTypesQuery(String biomartSpeciesName) {
+        return this.getDataURL() + microarrayTypesBaseQuery.replace("BIOMART_SPECIES_NAME", biomartSpeciesName);
+    }
+
+    /**
+     * Query Biomart for Microarray 'identifiers' associated with the microarray type and write the data to a file.
+     * @param biomartSpeciesName - String of species name in Biomart format (eg: hsapiens)
+     * @param biomartQueryFilePath - String, Path to the Biomart XML Query file
+     * @param biomartDataType - String, Type of data (either UniProt or Microarray) that will be queried for from BioMart
+     * @param fileSuffix - String that will be used to make filename (either _uniprot or _microarray) that will hold associated data.
+     * @throws IOException - Thrown when unable to write data to file.
+     */
+    private void queryBiomartAndStoreData(String biomartSpeciesName, String biomartQueryFilePath, String biomartDataType, String fileSuffix) throws IOException {
+
+        Set<String> biomartResponseLines = new HashSet<>();
+        try {
+            logger.info("Retrieving data associated with microarray type " + biomartDataType);
+            biomartResponseLines = queryBiomart(getBiomartIdentifierQuery(biomartQueryFilePath, biomartSpeciesName, biomartDataType), 0);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
```

Comment:
The exception here can be logged  in addition to being or instead of printed to STDOUT:

`logger.error("Unable to retrieve data associated with microarray type " + biomartDataType, e);`